synapse/cores/common.py

```diff
@@ -257,6 +257,31 @@ def _loadCoreModel(self):
         for item in self.getTufosByProp('syn:prop'):
             self._initPropTufo(item)
 
+    def addDataModel(self, modname, modl):
+        '''
+        Store all types/forms/props from the given data model in the cortex.
+        This should be done on initialization of an empty cortex only.
+
+        Example:
+
+            modname = 'synapse.models.foo'
+            core.addDataModel(modname, s_dyndep.tryDynFunc(modname, 'getDataModel'))
+        '''
+        vers = modl.get('version',0)
+        item = self.formTufoByProp('syn:model',modname,version=vers)
+
+        for name,info in modl.get('types',()):
+            self.formTufoByProp('syn:type',name,**info)
+
+        # load all forms after loading all types
+        for name,info,props in modl.get('forms',()):
+            self.formTufoByProp('syn:form',name,**info)
+
+            for prop,pnfo in props:
+                pnfo['form'] = name
+                fullprop = '%s:%s' % (name,prop)
+                self.formTufoByProp('syn:prop',fullprop,**pnfo)
+
```

Comment:
note that we didn't reuse this method to implement `_saveCoreModels()` because that routine first loads types, and then forms. seems this would minimize dependency issues, so its a better approach. but we can't replicate that here, since we only have access to one datamodel at a time.